sklearn/model_selection/_search_successive_halving.py

```diff
@@ -832,14 +844,26 @@ class HalvingRandomSearchCV(BaseSuccessiveHalving):
         - `None`: the `estimator`'s
           :ref:`default evaluation criterion <scoring_api_overview>` is used.
 
-    refit : bool, default=True
-        If True, refit an estimator using the best found parameters on the
-        whole dataset.
+    refit : bool or callable, default=True
+        Refit an estimator using the best found parameters on the whole
+        dataset.
+
+        Where there are considerations other than maximum score in
+        choosing a best estimator, ``refit`` can be set to a function which
+        returns the selected ``best_index_`` given ``cv_results_``. In that
+        case, the ``best_estimator_`` and ``best_params_`` will be set
+        according to the returned ``best_index_`` while the ``best_score_``
+        attribute will not be available.
 
         The refitted estimator is made available at the ``best_estimator_``
         attribute and permits using ``predict`` directly on this
         ``HalvingRandomSearchCV`` instance.
 
+        See
+        :ref:`sphx_glr_auto_examples_model_selection_plot_grid_search_refit_callable.py`
+        for an example of how to use ``refit=callable`` to balance model
+        complexity and cross-validated score.
```

Comment:
Small note, the title of this example is already "Balance model complexity and cross-validated score" - to avoid duplication, maybe we could do something like
```
see :ref:`this example <sphx_glr_auto_examples_model_selection_plot_grid_search_refit_callable.py>` for ...
```